x/bep3/types/swap.go

```diff
@@ -57,26 +80,41 @@ func (a AtomicSwap) Validate() error {
 	return nil
 }
 
-// NewAtomicSwap returns a new AtomicSwap
-func NewAtomicSwap(amount sdk.Coins, randomNumberHash cmn.HexBytes, expireHeight, timestamp int64, sender,
-	recipient sdk.AccAddress, senderOtherChain string, recipientOtherChain string, closedBlock int64, status SwapStatus) AtomicSwap {
-	return AtomicSwap{
-		Amount:              amount,
-		RandomNumberHash:    randomNumberHash,
-		ExpireHeight:        expireHeight,
-		Timestamp:           timestamp,
-		Sender:              sender,
-		Recipient:           recipient,
-		SenderOtherChain:    senderOtherChain,
-		RecipientOtherChain: recipientOtherChain,
-		ClosedBlock:         closedBlock,
-		Status:              status,
-	}
+// String implements stringer
+func (a AtomicSwap) String() string {
```

Comment:
Use raw string (`) instead of (") to avoid having to manually add new lines.  Same for `AssetSupply` string method. 
See https://github.com/Kava-Labs/kava/blob/42bd5e149f176c2031193bc5333f6c775f586cf8/x/cdp/types/cdp.go#L36 for example.

Reply:
After making the modifications the output result is:
```bash
"Atomic Swap:\n\t\t    ID:                       213FBD68B56B411CCB4D11BC1C36614105644D1DFE90581AE496BDA7EFD9AE68\n\t\t
  \   Status:                   Open\n\t\t    Amount:                   1111111bnb\n\t\t
  \   Random number hash:       095a454bb7c0dd88a766be0907e9ffcca470470d73a987150e49d7a707533129\n\t\t
  \   Expire height:            599\n\t\t    Timestamp:                1585154146\n\t\t
  \   Sender:                   kava1vry5lhegzlulehuutcr7nmdlmktw88awp0a39p\n\t\t
  \   Recipient:                kava1sl8glhaa9f9tep0d9h8gdcfmwcatghtdrfcd2x\n\t\t
  \   Sender other chain:       tbnb10uypsspvl6jlxcx5xse02pag39l8xpe7a3468h\n\t\t
  \   Recipient other chain:    tbnb1etjxt3t5r9899w3tf7dnktdue5ngu4ncdwckvt\n\t\t
  \   Closed block:             0\n\t\t    Cross chain:              true\n\t\t    Direction:
  \               Outgoing"
```

After updating line 148 of `client/cli/query.go` from `cdc.MustUnmarshalJSON(res, &atomicSwap).String()` to `cdc.MustUnmarshalJSON(res, &atomicSwap)`, the output result is:
```bash
swap: null
amount:
- denom: bnb
  amount: "1111111"
random_number_hash:
- 158
- 241
- 81
- 52
- 158
- 150
- 232
- 68
- 85
- 223
- 235
- 234
- 205
- 109
- 228
- 140
- 9
- 128
- 136
- 154
- 147
- 109
- 255
- 82
- 38
- 38
- 235
- 29
- 76
- 107
- 19
- 172
expire_height: 366
timestamp: 1585154354
sender: kava1sl8glhaa9f9tep0d9h8gdcfmwcatghtdrfcd2x
recipient: kava1sl8glhaa9f9tep0d9h8gdcfmwcatghtdrfcd2x
sender_other_chain: tbnb10uypsspvl6jlxcx5xse02pag39l8xpe7a3468h
recipient_other_chain: tbnb1etjxt3t5r9899w3tf7dnktdue5ngu4ncdwckvt
closed_block: 0
status: 1
cross_chain: true
direction: 1
```

For the 2nd option to work we'll need to return the string representation of the hex-encoded `random_number_hash` bytes, drop the `swap: null` interface from the results, and add an ID field. This would probably necessitate a new AugmentedAtomicSwap object. Worth noting that `status` and `direction` both have custom `UnmarshalJSON` methods that should return them as strings "Open" and "Incoming", but that isn't happening here.

I've already taken a stab at the string representation but I'm not sure how to get the `random_number_hash` string into the query results. The original string concatenation was a solution to get around this and the other requirements of an AugmentedAtomicSwap.

Reply:
fine to leave it as is then.

Reply:
`cliCtx.PrintOutput(atomicSwap)` defaults to printing the yaml encoded version of atomicSwap. So if you created a `MarshalYaml()` method for the random number hash type (or the swap type) you can control the output.